### PR TITLE
Update dep

### PR DESCRIPTION
**Pull request by SoManyHs**
Taken from https://github.com/golang/dep#installation

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -40,7 +40,7 @@ generate: $(SOURCES)
 
 .PHONY: generate-deps
 generate-deps:
-	go get github.com/tools/godep
+	curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
```
Comment (PettitWesley) [inline Makefile]: In order for this script to install the correct version (not the latest) I believe you need to set the `DEP_RELEASE_TAG`. That's what I had to do when I installed dep

Comment (SoManyHs) [reply Makefile]: done (with upgrade to 0.4.1 which is the latest anyway)

**Comment by SoManyHs** [comment]
`make` and `make test` pass.

`make generate-deps` installs dep v0.4.1
```
~/go/src/github.com/aws/amazon-ecs-cli(fix-makefile)$ make generate-deps
export DEP_RELEASE_TAG=$DEP_RELEASE_TAG
curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  4528  100  4528    0     0   6122      0 --:--:-- --:--:-- --:--:--  6118
ARCH = amd64
OS = darwin
Will install into /Users/hsinghui/go/bin
Release Tag = v0.4.1
Fetching https://github.com/golang/dep/releases/tag/v0.4.1..
Fetching https://github.com/golang/dep/releases/download/v0.4.1/dep-darwin-amd64..
Setting executable permissions.
Moving executable to /Users/hsinghui/go/bin/dep
go get github.com/golang/mock/mockgen
go get golang.org/x/tools/cmd/goimports

~/go/src/github.com/aws/amazon-ecs-cli(fix-makefile)$ ~/go/bin/dep version
dep:
 version     : v0.4.1
 build date  : 2018-01-24
 git hash    : 37d9ea0a
 go version  : go1.9.1
 go compiler : gc
 platform    : darwin/amd64
```

File: Makefile (hunk 1 above)
Comment (PettitWesley) [reply Makefile]: I'd recommend:

```
DEP_RELEASE_TAG=$DEP_RELEASE_TAG curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
```

That way the env var is only defined for one command.

Comment (PettitWesley) [reply Makefile]: Also, `$$` in needs to be `$`

Comment (SoManyHs) [reply Makefile]: Ah, thanks! fixed

File: Makefile [hunk 2]
```diff
@@ -40,7 +41,8 @@ generate: $(SOURCES)
 
 .PHONY: generate-deps
 generate-deps:
-	go get github.com/tools/godep
+	DEP_RELEASE_TAG=$DEP_RELEASE_TAG
+	curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
```
Comment (adnxn) [inline Makefile]: i don't think `curl ... | sh` is something we want to be doing in a make target.

**Comment by adnxn** [comment]
@SoManyHs, @PettitWesley, @allisaurus 
fwiw this is what we're doing for the agent repo: https://github.com/aws/amazon-ecs-agent/pull/1362

**Pull request merged**